src/index.js

```diff
@@ -28,22 +31,32 @@ function render() {
     `;
 }
 
-function postRender() {
-    ajaxView.postRender();
+function postRender(initPromise) {
+    ajaxView.postRender(initPromise);
 }
 
-let timeout = 1;
-const onTimeout = () => {
-    if (document.readyState === 'complete') {
-        const container = document.createElement('div');
-        container.innerHTML = render();
-        document.body.appendChild(container);
+function preInit(initPromise) {
+    dataView.preInit(initPromise);
+}
 
-        postRender();
-    }
-    else {
-        setTimeout(onTimeout, timeout *= 2);
+const init = new Promise(function(resolve, reject) {
+    preInit(init);
```

Comment:
Can you comment on what each of these side-effect functions do?

(same with line 43, 52, 54)

🙏 would be a huge help!